centreon-certified/omi/omi_events-apiv2.lua

```diff
@@ -0,0 +1,363 @@
+-- logfile (string): the log file to use
+-- loglevel (number): th log level (0, 1, 2, 3) where 3 is the maximum level
+-- port (number): the operation connector server port
+-- max_size (number): how many events to store before sending them to the server.
+-- max_age (number): flush the events when the specified time (in second) is reach (even if max_size is not reach).
+
+-- Libraries
+local curl = require("cURL")
+local http = require("socket.http")
+local ltn12 = require("ltn12")
+
+-- Centreon lua core libraries
+local sc_common = require("centreon-stream-connectors-lib.sc_common")
+local sc_logger = require("centreon-stream-connectors-lib.sc_logger")
+local sc_broker = require("centreon-stream-connectors-lib.sc_broker")
+local sc_event = require("centreon-stream-connectors-lib.sc_event")
+local sc_params = require("centreon-stream-connectors-lib.sc_params")
+local sc_macros = require("centreon-stream-connectors-lib.sc_macros")
+local sc_flush = require("centreon-stream-connectors-lib.sc_flush")
+
+-- workaround https://github.com/centreon/centreon-broker/issues/201
+local previous_event = ""
+
+--------------------------------------------------------------------------------
+-- EventQueue class
+--------------------------------------------------------------------------------
+
+local EventQueue = {}
+EventQueue.__index = EventQueue
+
+--------------------------------------------------------------------------------
+-- Constructor
+-- @param conf The table given by the init() function and returned from the GUI
+-- @return the new EventQueue
+--------------------------------------------------------------------------------
+function EventQueue.new(params)
+  local self = {}
+  self.fail = false
+
+  local mandatory_parameters = {
+      "ipaddr",
+      "url",
+      "port"
+  }
+
+  -- set up log configuration
+  local logfile = params.logfile or "/var/log/centreon-broker/omi_event.log"
+  local log_level = params.log_level or 2
+
+  -- initiate mandatory objects
+  self.sc_logger = sc_logger.new(logfile, log_level)
+  self.sc_common = sc_common.new(self.sc_logger)
+  self.sc_broker = sc_broker.new(self.sc_logger)
+  self.sc_params = sc_params.new(self.sc_common, self.sc_logger)
+
+  -- checking mandatory parameters and setting a fail flag
+  if not self.sc_params:is_mandatory_config_set(mandatory_parameters, params) then
+    self.fail = true
+  end
+
+  -- overriding default parameters for this stream connector if the default values doesn't suit the basic needs
+  self.sc_params.params.accepted_categories = params.accepted_categories or "neb"
+  self.sc_params.params.accepted_elements = params.accepted_elements or "service_status"
+  self.sc_params.params.source_ci = params.source_ci or "Centreon"
+  self.sc_params.params.ipaddr = params.ipaddr or "192.168.56.15"
+  self.sc_params.params.url = params.url or "/bsmc/rest/events/opscx-sdk/v1/"
+  self.sc_params.params.port = params.port or 30005
```

Comment:
if we put them as mandatory params, we shouldn't put default values. Personnally, i don't know this software so to me it looks more safe to let them be mandatory parameters and remove the default values